### DISCUSSION TITLE
Disable deprecation notices from command line

### DISCUSSION THREAD

**Discussion by RichardTMiles** [wp-cli/wp-cli#5765] [Show and tell]
If you've installed wp-cli using composer it is possible to invoke the command using: 

`php -d error_reporting=E_ALL&~E_DEPRECATED&~E_STRICT -f ./vendor/wp-cli/wp-cli/php/boot-fs.php`

You can extend this concept using composer scripts 

```
"scripts": {
    "wp:cli": "echo 'php -d error_reporting=E_ALL&~E_DEPRECATED&~E_STRICT -f ./vendor/wp-cli/wp-cli/php/boot-fs.php '",
    "wp:plugins:list:com": "$( composer run wp:cli ) plugin list --field=name --status=active --skip-plugins --url=example.com --skip-themes"
}
```

Happy Hacking, 
RM

**Comment by danielbachhuber** [comment]
Thanks for sharing, @RichardTMiles !

**Comment by pandabrand** [comment]
@RichardTMiles do you know of a way of doing this if you haven't used composer and just used the standard install?

**Reply by RichardTMiles** [reply]
The standard installation is just a `.phar` (https://www.php.net/manual/en/book.phar.php) file.  So probably just use the path `./wp-cli/php/boot-fs.php` where `wp-cli` is your executable (guess). I know `wp-cli` does some weird `exec` steps, so it's worth reading though the code to find the correct `bootstrapping` point, aka `boot-fs.php`. You can also install packages globally with composer using the `-g` flag. Also, I would like to advocate using composer at all cost (it's free). If you're not autoloading your PHP files using composer's [PSR-4](https://getcomposer.org/doc/04-schema.md) then you're not php-ing well!!! Python has `pip`, Node has `npm`, and we got `composer`. Do not be afraid like I once was; pretty much every language has `standard-ish` package manager. Shoot, even operating systems do ( Window = chocolaty, Mac = Brew, Fedora = pkg, Ubuntu = apt, etc.).

And on a more serious side-advocacy note: The `composer` `scripts` section is much better than `npm's` since you are allowed to have a script point to an array of commands with `composer.json` but not `package.json`. Inlining bash with ||&& and `if` statements stinks and is super hard to edit. If we had a command we needed to use a lot, never keep it in your notes; no, it gets put in the scripts section and generalized for every member of the team. Profit POV: Your scripts are now tracked, and you gave them pretty names everyone can reference and update without a DM.

**Reply by RichardTMiles** [reply]
https://github.com/CarbonORM/CarbonWordPressExample
